alphagov/paas-cf
# [#146463199] CDN certificate monitoring

## What

We want to alert when CDN certificates are about to expire.

## How to review

* Deploy from this branch with metrics enabled:

  ```
  SELF_UPDATE_PIPELINE=false ENABLE_DATADOG=true BRANCH=cdn-certificate-monitoring make dev pipelines
  ```
* Create CloudFront distribution(s) in the dev account.
* Check metrics start coming through in Datadog. You may want to manually change the polling interval in `paas-cf/tools/metrics/main.go` so it collects them every minute, otherwise you will be waiting an hour for each set of metrics to come through.
* Find creative ways to invalidate/expire the certificates and check the monitor is triggered.
* Check the credential rotation works by using the documentation in the related team manual pull request: https://github.com/alphagov/paas-team-manual/pull/178
* Remember to tear down the CloudFront distributions.

## Who can review

Anyone but me, @bandesz, or @46bit
